c/tskit/tables.c

```diff
@@ -5388,6 +5381,489 @@ ancestor_mapper_run(ancestor_mapper_t *self)
+    fprintf(out, "===\nEdge table\n==\n");
+    for (j = 0; j < self->tables->edges.num_rows; j++) {
+        fprintf(out, "L:%f, R:%f, P:%d, C:%d\n", self->tables->edges.left[j],
+            self->tables->edges.right[j], self->tables->edges.parent[j],
+            self->tables->edges.child[j]);
+    }
+    fprintf(out, "===\nNode table\n==\n");
+    for (j = 0; j < self->tables->nodes.num_rows; j++) {
+        fprintf(out, "ID:%f, Time:%f, Flag:%d\n", (double) j,
+            self->tables->nodes.time[j], self->tables->nodes.flags[j]);
+    }
+    fprintf(out, "==\nSample pairs\n==\n");
+    for (j = 0; j < 2 * self->num_pairs; j++) {
+        fprintf(out, "%i ", (int) self->pairs[j]);
+        if (j % 2 != 0) {
+            fprintf(out, "\n");
+        }
+    }
+    fprintf(out, "===\nAncestral map\n==\n");
+    for (j = 0; j < self->tables->nodes.num_rows; j++) {
+        fprintf(out, "Node %d: ", (int) j);
+        for (u = self->ancestor_map_head[j]; u != NULL; u = u->next) {
+            fprintf(out, "(%f,%f->%d)", u->left, u->right, u->node);
+        }
+        fprintf(out, "\n");
+    }
+    fprintf(out, "===\nIBD segments\n===\n");
+    for (j = 0; j < self->num_pairs; j++) {
+        fprintf(out, "Pair (%i, %i)\n", (int) self->pairs[2 * j],
+            (int) self->pairs[2 * j + 1]);
+        for (u = self->ibd_segments_head[j]; u != NULL; u = u->next) {
+            fprintf(out, "(%f,%f->%d)", u->left, u->right, u->node);
+        }
+        fprintf(out, "\n");
+    }
+}
+
+int TSK_WARN_UNUSED
+tsk_ibd_finder_run(tsk_ibd_finder_t *self)
+{
+    const tsk_edge_table_t *input_edges = &self->tables->edges;
+    int ret = 0;
+    size_t j;
+    tsk_id_t u;
+    tsk_id_t current_parent = -1;
+    size_t num_edges = input_edges->num_rows;
+    tsk_segment_t *seg;
+    tsk_segment_t *s;
+    double intvl_l, intvl_r, current_time;
+    bool parent_should_be_added = true;
+
+    for (j = 0; j < num_edges; j++) {
+
+        if (current_parent >= 0 && current_parent != input_edges->parent[j]) {
+            parent_should_be_added = true;
+        }
+
+        // Stop if the processed node's time exceeds the max time.
+        current_parent = input_edges->parent[j];
+        current_time = self->tables->nodes.time[current_parent];
+        if (current_time > self->max_time) {
+            goto out;
+        }
+
+        // if (current_parent == 4) {
+        //     printf("\nEdge: L: %f, R: %f, P: %i, C: %i\n", input_edges->left[j],
+        //     input_edges->right[j], (int) input_edges->parent[j], (int)
+        //     input_edges->child[j]);
+        // }
+
+        // Extract segment.
+        seg = tsk_ibd_finder_alloc_segment(
+            self, input_edges->left[j], input_edges->right[j], input_edges->child[j]);
+        // Create a SegmentList holding all of the sample segments descending from
+        // seg.
+        u = seg->node;
+        if (self->is_sample[u]) {
+            // printf("hereee");
```

Comment:
More prints - can you have a look through the diff for any more of these please?